docs/design/content-type-icon.md

```diff
@@ -0,0 +1,207 @@
+# Content Type icons
+
+* Author: Damien Pobel <dp@ez.no>
+* Created: 2016-05-20
+* JIRA: https://jira.ez.no/browse/EZP-23022
+* Status: Design Review
+
+## Context
+
+A Content Item is a kind of instance of a Content Type. The type of a Content
+Item is a very important information to provide to the user. One graphical way
+to carry this information is to use an icon near the Content Item name. This
+is especially useful when the available space is limited like while displaying
+the content tree. By essence, the Content Types are project specific so it's
+important to make that feature ease to configure and to extend for integrators.
+
+### How it worked in eZ Publish
+
+In eZ Publish, the Content Type icons were defined in an ini setting file
+called icons.ini. Here's an how this was written:
```

Comment:
Nitpick: it was called icon.ini :) https://github.com/ezsystems/ezpublish-legacy/blob/master/settings/icon.ini